### ISSUE TITLE
Smart crop add an extra space in `_formatted` field

### ISSUE THREAD

**Issue by alallema** [meilisearch/meilisearch#2528]
**Describe the bug**
The smart crop add an extra space in the `_formatted` field.

**To Reproduce**
1. Send a search request with this parameters
```
{
	"q": "to",
	"attributesToCrop": ["Title"],
	"cropLength": 2,
	"cropMarker": "(ꈍᴗꈍ)"
}
```
2. Get this result on the v0.28.0rc0
```
    "Title": "(ꈍᴗꈍ)Guide to (ꈍᴗꈍ)",
```
3. When I had this result with v0.27.1
```
    "Title": "(ꈍᴗꈍ)Guide to(ꈍᴗꈍ)",
```

**Expected behavior**
I'm not sure if this behavior is expected, but even if it is, it's more complicated to remove an extra space than to add one directly in the `cropMarker`.

**Meilisearch version:** v0.28.0rc0

**Comment by curquiza** [comment]
@ManyTheFish let us know if it's expected 😇

**Comment by ManyTheFish** [comment]
hey @curquiza and @alallema!
I think that the behavior expectation is fulfilled until we return the good amount of words.
However, I can force the new formater to trim spaces. 😄

**Comment by bidoubiwa** [comment]
With the crop marker it is clearer that it is a bit weird: 

`…book about a prince that walks …`

**Comment by curquiza** [comment]
Closed by ##2546
